ast/dml.go

```diff
@@ -276,6 +276,8 @@ type SelectField struct {
 	Expr ExprNode
 	// Alias name for Expr.
 	AsName model.CIStr
+	// Auxiliary stands for if this field is auxiliary.
```

Comment:
Not clear enough. Please add its use case. For example:
When we add a Field into SelectField list which is used for having/orderby clause but the field is not in projection clause, we should set its Auxiliary to true. Then the TrimExec will trim the field.